src/websocket.cpp

```diff
@@ -82,59 +153,90 @@ void app_websocket(void) {
 
   esp_websocket_client_start(client);
 
-  // Example sending data
-  // Livekit__AddTrackRequest r LIVEKIT__ADD_TRACK_REQUEST__INIT;
-  // size_t size = livekit__add_track_request__get_packed_size(&r);
-  // uint8_t *buffer =(uint8_t*) malloc(size);
-  // livekit__add_track_request__pack(&r,buffer);
-  // esp_err_t err = esp_websocket_client_send_bin(client,(char*)buffer,size,
-  // portMAX_DELAY); if (err != ESP_OK) {
-  //   printf("Failed to send data: %s\n", esp_err_to_name(err));
-  // } else {
-  //   printf("Data sent successfully\n");
-  // }
+  peer_init();
+  g_pc = peer_connection_create(&config);
+  peer_connection_oniceconnectionstatechange(g_pc, oniceconnectionstatechange);
+  peer_connection_ondatachannel(g_pc, onmessage, onopen, onclose);
+
 
+
+  xTaskCreatePinnedToCore(peer_connection_task, "peer_connection", 8192, NULL, 5, &xPcTaskHandle, 1);
+
+
+  int state = 0;
   while (true) {
+    if(eState == PEER_CONNECTION_NEW && state == 1) {
+      state=2;
+
+      // Example sending data
+      Livekit__SessionDescription r LIVEKIT__SESSION_DESCRIPTION__INIT;
+      r.sdp = peer_connection_get_offer(g_pc);
+      r.type = "offer";
+      size_t size = livekit__session_description__get_packed_size(&r);
+      uint8_t *buffer =(uint8_t*) malloc(size);
+      livekit__session_description__pack(&r,buffer);
+
+      esp_err_t err = esp_websocket_client_send_bin(
+        client,(char*)buffer,
+        size,
+        portMAX_DELAY); 
+      if (err != ESP_OK) {
+        printf("Failed to send data: %d\n", (int)(err));
+      } else {
+        printf("Data sent successfully\n");
+      }
```

Comment:
I am getting errors sending data, I am not sure if its because I am not sending the correct messages to livekit